quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/internal/core/QuarkusClasspathListenerManager.java

```diff
@@ -0,0 +1,90 @@
+package com.redhat.quarkus.jdt.internal.core;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.core.ElementChangedEvent;
+import org.eclipse.jdt.core.IElementChangedListener;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaElementDelta;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+
+public class QuarkusClasspathListenerManager {
+
+	private static final QuarkusClasspathListenerManager INSTANCE = new QuarkusClasspathListenerManager();
+
+	public static QuarkusClasspathListenerManager getInstance() {
+		return INSTANCE;
+	}
+
+	private static class QuarkusListener implements IElementChangedListener {
+
+		@Override
+		public void elementChanged(ElementChangedEvent event) {
+			List<String> projectsToUpdate = new ArrayList<String>();
+			processDelta(event.getDelta(), projectsToUpdate);
+			if (projectsToUpdate.size() > 0) {
+				JavaLanguageServerPlugin.getInstance().getClientConnection()
+						.executeClientCommand("quarkusTools.resetCache", projectsToUpdate);
+			}
+		}
+
+		private void processDelta(IJavaElementDelta delta, List<String> projectsToUpdate) {
+			IJavaElement el = delta.getElement();
+			switch (el.getElementType()) {
+			case IJavaElement.JAVA_MODEL:
+				processDeltaChildren(delta, projectsToUpdate);
+				break;
+			case IJavaElement.JAVA_PROJECT:
+				if (isCreatedOrDeleted(delta) || isClasspathChanged(delta.getFlags())) {
+					IJavaProject project = (IJavaProject) el;
+					projectsToUpdate.add(project.getProject().getName());

```

Comment:
I feel like it'd be safer to send project uris instead

Reply:
why? In Quarkus LS server side we have project names.

Reply:
Relying on the project name alone seems brittle. If I have projects under foo/foo and foo/barfoo, foo classpath changes, then barfoo would also be reset by the client.
QuarkusProjectInfo needs to hold the project uri

Reply:
project uri = IProject#getLocation().toOSString() ?